OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -502,15 +502,24 @@ public OpenAlAsyncLoadSound(uint source, bool looping, bool relative, WPos pos,
 		{
 			// Load a silent buffer into the source. Without this,
 			// attempting to change the state (i.e. play/pause) the source fails on some systems.
-			var silentSource = new OpenAlSoundSource(SilentData, channels, sampleBits, sampleRate);
+			var silentSource = new OpenAlSoundSource(SilentData, SilentData.Length, channels, sampleBits, sampleRate);
 			AL10.alSourcei(source, AL10.AL_BUFFER, (int)silentSource.Buffer);
 
 			playTask = Task.Run(async () =>
 			{
 				MemoryStream memoryStream;
 				using (stream)
 				{
-					memoryStream = new MemoryStream();
+					int? capacity = null;
+					try
+					{
+						capacity = (int)stream.Length;
+					}
+					catch (NotSupportedException)
+					{
```

Comment:
I think we use `catch (...) { }` elsewhere.

Reply:
Could we do something like
```csharp
try
{
	memoryStream = new MemoryStream((int)stream.Length);
}
catch (NotSupportedException)
{
	// Fallback for stream types that don't support .Length
	memoryStream = new MemoryStream();
}
```
for clarity?

Reply:
Done.